internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +147,74 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
-	return monitor, nil
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
+	if r.config.monitorConfig.MonitorConfigCore().IsCollectdBased() {
+		configureCollectdOnce.Do(func() {
+			r.logger.Info("Configuring collectd")
+			r.setUpCollectdConfig(extensions)
+			r.setUpEnvironment()
```

Comment:
I think `setUpEnvironment()` needs to happen even when collectd doesn't need to be started (which has a dependency on `setUpCollectdConfig()`) for sfxcollectd and gopsutil monitors.  Maybe via a new `configureEnvironmentOnce()`?

Reply:
Yes, that's a good point. Will decouple the setup of `SmartConfigProvider` from the collectd use-case.

Reply:
Updated.